query.go

```diff
@@ -1082,9 +1104,23 @@ func (s *ChainService) sendTransaction(tx *wire.MsgTx, options ...QueryOption) e
 						// request multiple times, we
 						// need to de-duplicate them
 						// using a map.
-						replies[sp.ID()] = struct{}{}
-
-						close(peerQuit)
+						replyChan := make(chan struct{})
+						replies[sp.ID()] = replyChan
+
+						// Okay, so the peer responded
+						// with an INV message, and we
+						// sent out the TX. If we never
+						// hear anything back from the
+						// peer it means they accepted
+						// the tx. If we get a reject,
+						// things are clear as well. But
+						// what if they are just slow to
+						// respond? We'll give them
+						// another bit of time to
+						// respond.
+						closeEventually(
```

Comment:
I think for this to be safe, we want to wrap it in a `sync.Once` to avoid a panic if we end up closing the channel twice.

Reply:
You're right. This is also a bit hard to understand, so I did a bit more refactoring and added a `delayedCloser` helper struct.